### PR TITLE
Add `--test-pattern` CLI option

### PR DESCRIPTION
**Pull request by tomek-szczesny**
Here's a little pattern generating code. It may have to be worked on because I'm not sure if it's sent to the printer faithfully. 

The bigger pattern indeed sends more data to the printer even though its physical size is the same, so I reckon this should work.

### CODE REVIEW

File: src/dymoprint/labeler.py [hunk 1]
```diff
@@ -145,10 +145,10 @@ def tapeColor(self, value):
...
         """Set the number of bytes sent in the following lines. (MLF)"""
 
-        if value < 0 or value + self.dotTab_ > self.max_bytes_per_line(
-            self.tape_size_mm
-        ):
-            raise ValueError
+        #if value < 0 or value + self.dotTab_ > self.max_bytes_per_line(
+        #    self.tape_size_mm
+        #):
+        #    raise ValueError
```
Comment (maresb) [inline src/dymoprint/labeler.py]: You can just delete this.

Interesting how `tape_size_mm` isn't actually doing anything. It's only used in `chainMark`, whatever that means. Let's leave that part in, just so `chainMark` isn't explicitly broken.

```suggestion
        self.tape_size_mm
```

File: src/dymoprint/command_line.py [hunk 2]
```diff
@@ -191,6 +197,10 @@ def main():
 
     bitmaps = []
 
+    if args.test_pattern:
+        args.text = " "
```
Comment (maresb) [inline src/dymoprint/command_line.py]: Ya, this is annoying, but not worth properly fixing in this PR. Does `""` work?

**Review by maresb — CHANGES_REQUESTED** [review]

File: src/dymoprint/command_line.py (hunk 2 above)
Comment (maresb) [reply src/dymoprint/command_line.py]: Ah, nevermind, I see that you're trying to do this as a solution to #88. (I was thinking this was about something else.)

CLI parameter validation happens in the Click library which is pretty idiosyncratic, so don't worry about it.

Comment (maresb) [inline src/dymoprint/command_line.py]:
```suggestion
```

**Comment by tomek-szczesny** [comment]
Before this gets pulled, I'd like to add some horizontal lines after the checkerboard. I've got a nice idea for it.

And perhaps --test-pattern argument should ignore all other arguments, because the following prints are obviously misaligned (canvas of a wrong size).

**Review by maresb — APPROVED** [review]

**Comment by maresb** [comment]
Ok, just let me know when you're ready. It's good enough for me. I can't be bothered at the moment to fix #88.

**Comment by tomek-szczesny** [comment]
I think this is ready for a pull.

**Comment by maresb** [comment]
For discoverability, I'd like to do the matrix optimization removal in a separate PR. (I'm thinking that at some point in the future if we support huge labels then it'd be nice to have a clean reference to the point when that code was removed.)

That stuff can be super-annoying if you're not a frequent Git user. Would you like me to crank that out for you?

**Comment by tomek-szczesny** [comment]
> For discoverability, I'd like to do the matrix optimization removal in a separate PR. (I'm thinking that at some point in the future if we support huge labels then it'd be nice to have a clean reference to the point when that code was removed.)
> 
> That stuff can be super-annoying if you're not a frequent Git user. Would you like me to crank that out for you?

I would appreciate that, thanks. 

But I'm not sure if it makes any difference - it is a separate commit after all. You can pull in such a way that the whole pull request isn't seen as a single commit. I think I've done that once. :)

**Comment by maresb** [comment]
> But I'm not sure if it makes any difference - it is a separate commit after all. 

This is true except for the automatic release notes, which are on a per-PR basis, and that's what I'm aiming for.

> You can pull in such a way that the whole pull request isn't seen as a single commit.

Ya, that's the squash-merge. I'll do it here.

**Comment by tomek-szczesny** [comment]
Sorry, I just found one last bug. ;)

The horizontal line pattern was one dot off.

**Comment by maresb** [comment]
@tomek-szczesny, I cherry-picked your commit to a branch `remove-dottab-matrix-opt`. Are you able to open a PR for it? For me it works via this link:
https://github.com/computerlyrik/dymoprint/compare/remove-dottab-matrix-opt?expand=1

**Comment by tomek-szczesny** [comment]
@maresb sorry I can't. I have to be a collaborator, the error says.

**Pull request merged**